src/components/BuyManaWithFiatModal/BuyManaWithFiatModal.tsx

```diff
@@ -0,0 +1,196 @@
+import * as React from 'react'
+import ModalContent from 'semantic-ui-react/dist/commonjs/modules/Modal/ModalContent'
+import Icon from 'semantic-ui-react/dist/commonjs/elements/Icon/Icon'
+import { Network } from '@dcl/schemas'
+import { Modal } from '../Modal/Modal'
+import { ModalNavigation } from '../ModalNavigation/ModalNavigation'
+import { Loader } from '../Loader/Loader'
+import {
+  BuyWithFiatNetwork,
+  NetworkGatewayProps,
+  BuyWithFiatNetworkProps,
+  networksNames
+} from './Network'
+import './BuyManaWithFiatModal.css'
+import './Network.css'
+
+type SelectedNetwork = BuyManaWithFiatModalNetworkProps &
+  BuyWithFiatNetworkProps
+
+export type BuyManaWithFiatModalProps = {
+  className?: string
+  message?: React.ReactNode
+  hasError?: boolean
+  open?: boolean
+  loading?: boolean
+  i18n?: BuyManaWithFiatModalI18N
+  networks: (BuyManaWithFiatModalNetworkProps & BuyWithFiatNetworkProps)[]
+  onClose?: () => void
+  onInfo?: () => void
+}
+
+export type BuyManaWithFiatModalNetworkProps = {
+  type: Network
+  i18n?: BuyManaWithFiatModalNetworkI18N
+  gateways: Omit<NetworkGatewayProps, 'network'>[]
+  onClick?: () => void
+}
+
+export type BuyManaWithFiatModalI18N = {
+  title: React.ReactNode
+  subtitle: React.ReactNode
+  error: React.ReactNode
+}
+
+export type BuyManaWithFiatModalNetworkI18N = {
+  cta: React.ReactNode
+  ctaSubtitle: React.ReactNode
+}
+
+class BuyManaWithFiatModalNetwork extends React.PureComponent<BuyManaWithFiatModalNetworkProps> {
+  render(): JSX.Element {
+    const { type, i18n, onClick } = this.props
+
+    const cta: React.ReactNode = `${networksNames[type]} MANA`
+    let ctaSubtitle: React.ReactNode = ''
+
+    switch (type) {
+      case Network.MATIC:
+        ctaSubtitle = 'Use it to buy most wearables and emotes.'
+        break
+
+      case Network.ETHEREUM:
+        ctaSubtitle = 'Use it to buy LAND, names and specific wearables.'
+        break
+    }
+
+    return (
+      <div
+        className={`dcl option ${type.toLowerCase()}`}
+        onClick={onClick}
+        key={type}
+      >
+        <div className="image" />
+        <div className="info">
+          <div className="cta">{i18n?.cta || cta}</div>
+          <div className="ctaSubtitle">{i18n?.ctaSubtitle || ctaSubtitle}</div>
+        </div>
+        <Icon name="chevron right" />
+      </div>
+    )
+  }
+}
+
+export class BuyManaWithFiatModal extends React.Component<BuyManaWithFiatModalProps> {
+  state: Readonly<{
+    selectedNetwork: SelectedNetwork
+  }> = {
+    selectedNetwork: null
+  }
+
+  static defaultProps = {
+    className: '',
+    hasError: false,
+    loading: false,
+    i18n: {
```

Comment:
To be honest, I thought about using i18n to maintain consistency between other components in the UI.

For example, the `LoginModal`: 
https://github.com/decentraland/ui/blob/02e4041ac2892a1902dabf27dda2d5370b66d093/src/components/LoginModal/LoginModal.tsx#L126-L136

Why `texts` instead of the name of the props directly (title, subtitle, etc., as component props), in your opinion?

Reply:
Fair enough 😄